### ISSUE TITLE
State transitions upon witnessing tx `in mempool`

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#206]
## TLDR

Add support to transition the CFD of the party that did not trigger the tx publication into pending state upon witnessing a tx in mempool. 

## TLR

At the moment, when one party publishes a transaction we only transition this party into a `pending` state until the transaction was confirmed. 
The other party only notices a change in the UI once we reach finality - which can be significant time. 
This is not an issue per-se, because if the other party tries to publish *the same* action it will transition. However, it can have weird side effects. E.g. actions that should not be a available anymore are still available and may lead to errors. 
Example: Maker publishes commit. Taker tries to go for collaborative settlement which will be declined. The taker could have already known that that's not possible anymore.

**Comment by DeliciousHair** [comment]
Incorporating into #357

**Comment by da-kami** [comment]
I think this ticket should not be in scope at the moment. We should tackle this once the state machine was refactored. 

This ticket is not super important - it does give the user more immediate feedback, but without that feedback there is only few bugs or side effects (like triggering actions that are potentially not available anymore...). This becomes even less important as we removed the actions in question (rollover) in the new UI. The new UI only has `Close` and the maker is assumed to not force close at the moment. 

This ticket has quite a rat-tail that is: If we transition to "in mempool" states that don't involve confirmation we also have to make sure to retry publishing transations  in such states (...)

**Comment by bonomat** [comment]
This project is unmaintained
